astroid/interpreter/_import/util.py

```diff
@@ -72,6 +74,11 @@ def is_namespace(modname: str) -> bool:
 
         # Update last_submodule_search_locations
         if found_spec and found_spec.submodule_search_locations:
+            if any(
+                location in EXT_LIB_DIRS
+                for location in found_spec.submodule_search_locations
+            ):
+                return False
```

Comment:
Look like this line is not covered, is there something wrong with the tests ?

Reply:
Yes! b89ef27871d01a28e6e5ece6458836e0e8580b4f

(After testing my test, I butchered it when trying to "simplify" it further.)

Reply:
Can we add a comment about this. This function is becoming very intricate really quickly and I want to avoid the guessing we're doing for the old namespace detection in a couple of years from now.